pkg/values/map.go

```diff
@@ -185,7 +185,9 @@ func unwrapsValues(f reflect.Type, t reflect.Type, data any) (any, error) {
 		n := reflect.New(t).Interface()
 		err := dv.UnwrapTo(n)
 		if err != nil {
-			return nil, err
+			// Do not return the error to allow mapstructure to retry with different types
```

Comment:
what happens when those retries fail? is the ultimate error swallowed or is it returned and intelligible?

Reply:
Then mapstructure will try to map it with the type that it has and would fail anyways.